core/domain/exp_services.py

```diff
@@ -1926,9 +2120,24 @@ def are_changes_mergeable(exp_id, frontend_version, change_list):
         }
 
         if len(added_state_names) > 0 or len(deleted_state_names) > 0:
-            # Here we will send the changelist, version, latest_version,
-            # and exploration to the admin, so that the conditions
-            # can be reviewed.
+            # In case of the addition and the deletion of the state,
+            # we are rejecting the mergebility because these cases
+            # change the flow of the exploration and are quite complex
+            # for now to handle. So in such cases, we are sending the
+            # changelist, frontend_version, backend_version and
+            # exploration id to the admin, so that we can look into the
+            # situations and can figure out the way if it’s possible to
+            # handle these cases.
```

Comment:
Optional: Addition/deletion of state should not affect changes related to translation, right? (Also, it should only affect the answer groups, right? (not sure!))

Reply:
Actually, it may affect and it may not affect. At present we don't have a way to know what else changed in an answer group like maybe the feedback is also changed when adding/deleting the state so we cannot let translations being added in such cases. Also, it may be a case that rule is changed in the answer group along with the addition and deletion of the so that affects the solution. And we cannot make changes in the interaction id if someone has changed the answer group. So these properties are interrelated and are complex to handle now, that's why we are rejecting the changes in these cases. So Sean told me to send the change list to the admin in such cases so that they can be reviewed and if possible can be handled in the future. I have covered all the scenarios in my proposal. You can take a look at it [here](https://docs.google.com/document/d/1Wu2ZKAEhUdi95P3zwiji4PYPsMdlQeRGMBu8IGxY3eY/edit?usp=sharing).